From b28e2129be268b801568a047ce7dea39a473425a Mon Sep 17 00:00:00 2001
From: Hiren Vadukul <hirenv@ext.mechasystems.com> 
Date: Mon, 1 Sep 2025 18:43:48 +0530
Subject: [PATCH] Added the expander node & enable the haptic motor

Added the gpio expander node and set the haptic
motor nSleep pin always high.

Signed-off-by: Hiren Vadukul <hirenv@ext.mechasystems.com>

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts b/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts
index 9f6c153dac24..ccd97e9e7c77 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts
@@ -454,6 +454,24 @@ &i2c6 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c6>;
 	status = "okay";
+
+	pca9535:gpio@20 {
+	        compatible = "nxp,pca9535";
+	        reg = <0x20>;
+	        gpio-controller;
+	        #gpio-cells = <2>;
+	        interrupt-parent = <&gpio4>;
+	        interrupts = <28 IRQ_TYPE_LEVEL_LOW>;
+	        interrupt-controller;
+	        #interrupt-cells = <2>;
+
+		nSleep {
+	                gpio-hog;
+	                gpios = <15 GPIO_ACTIVE_HIGH>;
+	                output-high;
+	                line-name = "haptic motor nSleep";
+		};
+	};
 };
 
 &uart1 { 
-- 
2.25.1

